mock/src/main.rs

```diff
@@ -0,0 +1,137 @@
+    let data = unsafe { FieldElement::bytes_as_elements(&seed_bytes) }.unwrap();
+    let seed: Word = [data[0], data[1], data[2], data[3]];
+
+    seed
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    env_logger::init();
+
+    let args = Args::parse();
+
+    let small_rng = match args.seed {
+        Some(seed) => Pcg64::seed_from_u64(seed),
+        None => Pcg64::from_entropy(),
+    };
+
+    println!(
+        "Generating mock data, results will be saved to {}",
+        args.output.to_str().expect("invalid file name")
+    );
+
+    let mut output = File::create(args.output)?;
+
+    // Block 1
+    // --------------------------------------------------------------------------------------------
+    let mut mock_chain = MockChain::new(small_rng);
+    let start = Instant::now();
+
+    let faucet = mock_chain.build_fungible_faucet_with_seed(
+        seed_to_word("3ef982ea3dca3f89179e1e86ef1c263896ef1e07e76325ce7c69825046bf75ec"),
+        OnChain::Yes,
+        DEFAULT_ACCOUNT_CODE,
+        Digest::default(),
+    );
+    println!("Fungible faucet created {} [took: {}s]", faucet, start.elapsed().as_secs());
+    mock_chain.seal_block();
+
+    // Block 2
+    // --------------------------------------------------------------------------------------------
+    let amount = 100;
+    let asset = mock_chain.build_fungible_asset(0, amount);
+
+    let storage = vec![];
+    let assets = vec![asset];
+    let start = Instant::now();
+    let account0 = mock_chain.build_account_with_seed(
+        seed_to_word("fbee7d33ae8550dfde6d63ad899177d42382f8a7f04b55a5a57ee6a30a2c2a52"),
+        DEFAULT_ACCOUNT_CODE,
+        storage,
+        assets,
+        Immutable::No,
+        OnChain::No,
+    );
+    println!("Account created {} [took: {}s]", account0, start.elapsed().as_secs());
+
+    let storage = vec![];
+    let assets = vec![];
+    let start = Instant::now();
+    let account1 = mock_chain.build_account_with_seed(
+        seed_to_word("0a51501ec59c8da792ee0393778653c8b4be16e1a04bef64d7aa46ec5bd27a3c"),
+        DEFAULT_ACCOUNT_CODE,
+        storage,
+        assets,
+        Immutable::No,
+        OnChain::No,
+    );
+    println!("Account created {} [took: {}s]", account1, start.elapsed().as_secs());
+
+    let storage = vec![];
+    let assets = vec![];
+    let start = Instant::now();
+    let account2 = mock_chain.build_account_with_seed(
+        seed_to_word("f4f854256335c210732d0ee54ed73d3cc14cb6db4dfd580339f75ff819634d29"),
+        DEFAULT_ACCOUNT_CODE,
+        storage,
+        assets,
+        Immutable::No,
+        OnChain::No,
+    );
+    println!("Account created {} [took: {}s]", account2, start.elapsed().as_secs());
+
+    mock_chain.seal_block();
+
+    // Serialize
+    // --------------------------------------------------------------------------------------------
+    let data = postcard::to_allocvec(&mock_chain)?;
+
+    output.write_all(&data)?;
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
minor nit: let's add our usual section separator above this to indicate the start of the `TESTS` section.

Reply:
added